akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -341,8 +344,8 @@ abstract class StatefulStage[In, Out] extends PushPullStage[In, Out] {
    */
   abstract class State extends StageState[In, Out]
 
-  private var emitting = false
-  private var _current: StageState[In, Out] = _
+  private[this] var emitting = false
+  private[this] var _current: StageState[In, Out] = _
```

Comment:
ok, the extra private method can't matter for binary compatibility, it must have been another scenario I was thinking of.
Feel free to keep the `[this]`.
(I bet that hotspot will do it's job though)